lib/krb5/acache.c

```diff
@@ -557,6 +557,11 @@ acc_resolve_2(krb5_context context, krb5_ccache *id, const char *res, const char
     }
 
     error = (*a->context->func->open_ccache)(a->context, res, &a->ccache);
+    if (error == ccErrCCacheNotFound) {
+        a->ccache = NULL;
+        a->cache_name = NULL;
```

Comment:
Do we not need to also free(s); here?

Reply:
for a->cache_name? a->ccache presumably is NULL in this case?

Reply:
I don't think so.` a->ccache` is NULL because `open_ccache()` failed. `a->cache_name` is set by `get_cc_name()` which is not yet called. Or did I misunderstand?

Reply:
Which makes me wonder if it's necessary to set either of these to NULL?

Reply:
Also, should we be calling `get_kdc_time_offset()` in the `ccErrCCacheNotFound` case? If so, this logic should be moved further up.

Reply:
s may have been allocated by the asprintf() in the if (sub) block above. If we return without freeing s, it will be leaked.